### ISSUE TITLE
question about softmax_cross_entropy_loss_layer

### ISSUE THREAD

**Issue by BrettLL** [yjxiong/caffe#171]
In this layer, its backward is bottom_diff = p(i) - q(i), but i get the result: bottom_diff = p(i)*(q(i) - 1),  am I wrong and could you give me the derivation procedure about the kl-distance backward?? thanks~

**Comment by BannyStone** [comment]
suppose p is the distribution of teacher and q is the distribution of student
![w tsvj qoj8 x nzbxj248](https://user-images.githubusercontent.com/29084651/27132883-02ea2a70-5143-11e7-8228-a14ca03507da.png)

**Comment by yjxiong** [comment]
@BrettLL Please be noted that this layer includes both Softmax normalization and cross entropy computation. Thus the gradients are with respect to the unnormalized input `z_k`, not the normalized probability `q_k`.  The gradient equation can be written as `q_k - p_k` as @BannyStone 
 illustrated above. Because `q_k` is already computed in the forward process, this simplifies the backward computation.

@BannyStone Thanks for the derivations.

**Comment by BrettLL** [comment]
@BannyStone Thanks for your derivations and @yjxiong Thanks for your explanation. I got it~~
